pepper-apis/src/main/java/org/broadinstitute/ddp/route/AdminCreateUserLoginAccountRoute.java

```diff
@@ -0,0 +1,178 @@
+            }
+
+            if (user.hasAuth0Account()) {
+                LOG.error("Attempted to create login account for user with an existing auth0 account, userGuid={}, studyGuid={}",
+                        userGuid, studyGuid);
+                throw ResponseUtil.haltError(HttpStatus.SC_BAD_REQUEST,
+                        new ApiError(ErrorCodes.OPERATION_NOT_ALLOWED, "User already has login account"));
+            } else if (user.isTemporary()) {
+                LOG.error("Attempted to create login account for temporary user, userGuid={}, studyGuid={}", userGuid, studyGuid);
+                throw ResponseUtil.haltError(HttpStatus.SC_BAD_REQUEST,
+                        new ApiError(ErrorCodes.OPERATION_NOT_ALLOWED, "User is a temporary user"));
+            }
+
+            Auth0ManagementClient auth0Mgmt = createManagementClient(handle, studyGuid);
+            String auth0ClientId = payload.getAuth0ClientId();
+
+            // Find database connection to sign up new user
+            var listResult = auth0Mgmt.listClientConnections(auth0ClientId);
+            if (listResult.hasThrown() || listResult.hasError()) {
+                Exception e = listResult.hasThrown() ? listResult.getThrown() : listResult.getError();
+                LOG.warn("Error getting client connections for auth0 client {}", auth0ClientId, e);
+                throw ResponseUtil.haltError(HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                        new ApiError(ErrorCodes.SERVER_ERROR, "Error looking up client"));
+            }
+
+            Connection dbConn = listResult.getBody().stream()
+                    .filter(conn -> conn.getStrategy().equals(Auth0ManagementClient.DB_CONNECTION_STRATEGY))
+                    .findFirst()
+                    .orElse(null);
+            if (dbConn == null) {
+                LOG.error("Client {} does not have database connection", auth0ClientId);
+                throw ResponseUtil.haltError(HttpStatus.SC_BAD_REQUEST,
+                        new ApiError(ErrorCodes.OPERATION_NOT_ALLOWED, "Client has no database connection"));
+            } else {
+                LOG.info("Using database connection with name {} for client {}", dbConn.getName(), auth0ClientId);
+            }
+
+            // Create auth0 user
+            String email = payload.getEmail();
+            String randomPassword = GuidUtils.randomPassword();
+            var userResult = auth0Mgmt.createAuth0User(dbConn.getName(), email, randomPassword);
+            if (userResult.hasThrown() || userResult.hasError()) {
+                Exception e = userResult.hasThrown() ? userResult.getThrown() : userResult.getError();
+                LOG.error("Error creating new auth0 account for user {}", user.getGuid(), e);
+                throw ResponseUtil.haltError(HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                        new ApiError(ErrorCodes.SERVER_ERROR, "Error setting up login account"));
+            } else {
```

Comment:
Consider finding out what a bad password error looks like and including a a couple of retries. Gonna happen that Auth0 will not like one of the passwords that have been autogenerated.

Reply:
I have tried the letters in the alphabet used here, so I know those work. I don't see any settings in Auth0 password policies that will restrict what letters you can use, so I think the alphabet is good. The only issues I see here is if a study had set a minimum length is that greater than what we use here, or we somehow generate password with 2+ same consecutive letters AND study has that setting turned on. In those cases, I see retries will definitely help (for min length, probably need to figure out a way to bump up the length).

Reply:
Added same checks and retries which should be sufficient for now.